zeroeq/detail/context.h

```diff
@@ -0,0 +1,21 @@
+
+/* Copyright (c) 2017, Human Brain Project
+ *                     Stefan.Eilemann@epfl.ch
+ */
+
+#pragma once
+
+#include <memory>
+
+namespace zeroeq
+{
+namespace detail
+{
+using ContextPtr = std::shared_ptr<void>;
+
+/** @return the current ZeroMQ context. Users need to hold onto this context to
+ *          extend its lifetime to all sockets created from the context.
```

Comment:
This documentation is not precise. The context is created if it doesn't exist.

This function is a bit strange for such a simple name as getContext.
Why storing a weak reference to the context internally? It seems as if you wanted to guarantee that all sockets will share the same context implicitly as long there's one object holding a context and remove the context once the last object using it is destroyed. If that's the intention it should be documented somewhere. I would rename it to getOrCreateSharedContext()